pkgs/tools/package-management/pdm/default.nix

```diff
@@ -31,15 +33,23 @@ in
 with python.pkgs;
 buildPythonApplication rec {
   pname = "pdm";
-  version = "2.10.0";
+  version = "2.10.1";
   format = "pyproject";
   disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-ziJJWVr59hsJJqCJljLfSbHHESYegFak+uFLU/k9kZM=";
+    hash = "sha256-0WZTHGWfxJBZM1RlRN0uFs9kjCum2JjIISatakIReoE=";
   };
 
+  # if we use `copy2` to copy files with permission,
+  # we will not be able to write to them because of the read-only nix store
+  # https://github.com/NixOS/nixpkgs/issues/265883
+  postPatch = ''
+    substituteInPlace src/pdm/cli/templates/__init__.py \
+      --replace "shutil.copy2" "shutil.copyfile"
+  '';
```

Comment:
Please check with frostming, whether we can get this upstreamed.

Reply:
reported in https://github.com/pdm-project/pdm/issues/2385

Reply:
It looks like the PR has already been submitted and merged. https://github.com/pdm-project/pdm/pull/2379
Thanks @adisbladis.

Reply:
No worries =)